Kotlin:Strings .lowercase() (#3794)
Codecademy/docs: content/kotlin/concepts/strings/terms/lowercase/lowercase.md

---
Title: '.lowercase()'
Description: 'The lowercase method in Kotlin is used to convert the characters of a string to lowercase.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Lowercase'
  - 'Strings'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

The `lowercase()` method in Kotlin is used to convert the characters of a string to lowercase. 

## Syntax

```pseudo
fun String.lowercase(locale: Locale = Locale.getDefault()): String
```

### Parameters

- `locale` (optional): A `Locale` object representing the language and region-specific rules for case conversion. The default value is `Locale.getDefault()`.

## Examples
Let's dive into some examples to see how the `.lowercase()` method can be used:

```
println("Iced frappé!".lowercase()) // iced frappé!
```

The output of this code will be:

```shell
hello, world!
```

Let's take a look at an example where we specify the locale:

> ```pseudo
> println("KOTLIN".lowercase()) 
> val turkishLocale = Locale.forLanguageTag("tr")
> println("KOTLIN".lowercase(turkishLocale))
> ```

Reviewer: ```suggestion
```kotlin
import java.util.Locale

fun main() {
	println("KOTLIN".lowercase()) 
	val turkishLocale = Locale.forLanguageTag("tr")
	println("KOTLIN".lowercase(turkishLocale)) 
}

```
```
To work with locales in Kotlin, you should use the java.util.Locale class, which is a part of the Java standard library.

---
Title: '.lowercase()'

> Description: 'The lowercase method in Kotlin is used to convert the characters of a string to lowercase.'

Reviewer 2: ```suggestion
Description: 'Converts the characters of a string to lowercase.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The `.lowercase()` method in Kotlin is used to convert the characters of a string to lowercase. 

Reviewer 2: ```suggestion
The **`.lowercase()`** method in Kotlin is used to convert the characters of a string to lowercase. 
```

## Syntax

```pseudo

> fun String.lowercase(locale: Locale = Locale.getDefault()): String

Reviewer 2: ```suggestion
String.lowercase(locale)
```

```

- `locale` (optional): A `Locale` object representing the language and region-specific rules for case conversion. The default value is `Locale.getDefault()`.

## Examples

Let's dive into some examples to see how the `.lowercase()` method can be used:

```kotlin
fun main() {

>     println("Iced frappé!".lowercase())

Reviewer 2: This example is a direct copy of an example in the Kotlin documentation. This is considered plagiarism, and is not permitted within docs. All docs content must be original. Revise the example with an example you have developed

}
```

The output of this code will be:

```shell
iced frappé!
```

Let's take a look at an example where we specify the locale:

```kotlin

> import java.util.Locale

Reviewer 2: This example demonstrates the use of the optional parameter, but it is not a case in which setting the locale results in a change. It would be better to use a case in which there was a different output.

---
Title: '.lowercase()'
Description: 'The lowercase method in Kotlin is used to convert the characters of a string to lowercase.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:

>  

Reviewer: ```suggestion
```
